pkg/reconciler/ingress/resources/gateway.go

```diff
@@ -158,18 +232,20 @@ func GatewayServiceNamespace(ingressGateways []config.Gateway, gatewayName strin
 	return "", fmt.Errorf("no Gateway configuration is found for gateway %s", gatewayName)
 }
 
-// getAllGatewaySvcNamespaces gets all of the namespaces of Istio gateway services from context.
-func getAllGatewaySvcNamespaces(ctx context.Context) []string {
+func getIngressGatewaySvcNameNamespaces(ctx context.Context) []metav1.ObjectMeta {
 	cfg := config.FromContext(ctx).Istio
-	namespaces := sets.String{}
+	nameNamespaces := []metav1.ObjectMeta{}
 	for _, ingressgateway := range cfg.IngressGateways {
 		// serviceURL should be of the form serviceName.namespace.<domain>, for example
 		// serviceName.namespace.svc.cluster.local.
-
+		name := strings.Split(ingressgateway.ServiceURL, ".")[0]
```

Comment:
you're splitting the same string twice. That feel redundant.
Also probably `SplitN` is more suitable.

Reply:
Done